run.go

```diff
@@ -139,38 +135,102 @@ var commandInspect = cli.Command{
 	Action: actionInspect,
 }
 
-func guessType(filename string) string {
-	switch {
-	case strings.Contains(filename, "://"):
-		return TypeURL
-	case strings.HasSuffix(filename, ".js"):
-		return TypeJS
-	default:
-		return ""
+func looksLikeURL(str []byte) bool {
+	// Try to parse string as URL
+	u, err := url.Parse(strings.TrimSpace(string(str)))
```

Comment:
Sorry, I don't understand. Why wouldn't we want to check for a supported scheme?  Also, I'm not sure what url.Parse() does but I'm betting it has more failure conditions than just a bad scheme prefix?  Isn't it safer to use that than a regex?